modules/repository/branch.go

```diff
@@ -71,7 +71,9 @@ func CreateNewBranch(doer *models.User, repo *models.Repository, oldBranchName,
 	}
 
 	if !git.IsBranchExist(repo.RepoPath(), oldBranchName) {
-		return fmt.Errorf("OldBranch: %s does not exist. Cannot create new branch from this", oldBranchName)
+		return models.ErrBranchDoesNotExist{
```

Comment:
Have you checked every call to CreateNewBranch and made sure they handle your new error correctly?

Reply:
The other place where CreateNewBranch is used is in **routers/repo/branch.go**  CreateBranch function. 

It does not "models.isErr" it but treat it as a ServerErrir. Should we add a check there as well ? If so, should we do it in this PR or in another CL ?